regression: 💩 [v0.2.0] panic: runtime error: index out of range [0] with length 0

Inside the root's working directory, `gh tp` runs fine. If I run it in another directory, it returns this error message. I believe it is related to  where I changed the ordering in `cmd/root.go`. 

I originally had

```go
if len(args) == 0 {

} else if args[0] == "-" {

}
```

But I changed it to 

```go
if args[0] == "-" {

} else if len(args) == 0 {

}
```

```bash
gh tp
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/esacteksab/gh-tp/cmd.init.func1(0x9df5a0, {0xa07f60, 0x0, 0x722fd2?})
        /home/bmorriso/local-repo/gh-tp/cmd/root.go:101 +0x498
github.com/spf13/cobra.(*Command).execute(0x9df5a0, {0xc0000120b0, 0x0, 0x0})
        /home/bmorriso/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1019 +0xa7b
github.com/spf13/cobra.(*Command).ExecuteC(0x9df5a0)
        /home/bmorriso/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1148 +0x40c
github.com/spf13/cobra.(*Command).Execute(...)
        /home/bmorriso/go/pkg/mod/github.com/spf13/cobra@v1.9.1/command.go:1071
github.com/esacteksab/gh-tp/cmd.Execute()
        /home/bmorriso/local-repo/gh-tp/cmd/root.go:224 +0x1a
main.main()
        /home/bmorriso/local-repo/gh-tp/gh-tp.go:8 +0xf
```

--- a/README.md
+++ b/README.md
@@ -25,16 +25,11 @@ To create a plan and the markdown from that plan, run
 
 ```bash
 gh tp
-Using config file: /Users/tempuser/.tp.toml
+✔  Markdown Created...
+✔  Plan Created...
 ```
 
-Two files will be created, the first an output file named, what you defined for the value of `planFile` in `.tp` config and a Markdown file named what you defined for the value of the parameter `mdFile` in the `.tp` config file.
-
-```bash
-ls |grep plan
-plan.md
-plan.out
-```
+Two files will be created, the first an output file named, what you defined for the value of `planFile` in `.tp.toml` config and a Markdown file named what you defined for the value of the parameter `mdFile` in the `.tp.toml` config file.
 
 ### Create Commit
 
@@ -68,7 +63,7 @@ If you're targeting a resource, you can still create markdown from that plan's o
 terraform plan -out plan.out -no-color  | gh tp -
 ```
 
-Like with `gh tp` two files will exist. The first being whatever you passed to `-out` for the file name in the above example (`plan.out` in the example above) and the Markdown file named whatever you defined as the value for the `mdFile` parameter in the `.tp` config file. `tp` does not create an additional plan having been passed the plan from `stdin`.
+Like with `gh tp` two files will exist. The first being whatever you passed to `-out` for the file name in the above example (`plan.out` in the example above) and the Markdown file named whatever you defined as the value for the `mdFile` parameter in the `.tp.toml` config file. `tp` does not create an additional plan having been passed the plan from `stdin`.
 
 ### Extended Example
 
@@ -82,7 +77,7 @@ The above example is intended to be just enough to get you started. If you'd lik
 <!-- markdownlint-disable-line MD028 -->
 
 > [!WARNING]
-> **_Pre 1.0.0 Alert_**. This is a new extension and as a result, the API isn't set yet. There have been two prototypes so far. The experiences from both have shaped what exists today, but there is still quite a bit left to do. So as I continue down the path of figuring things out, things are almost certainly going to change. **_Expect_** breaking changes in the early releases. I will _strive_ not to publish broken builds and will lean into Semver to identify `alpha`, `beta` and pre-releases.
+> > **_Pre 1.0.0 Alert_**. This is a new extension and as a result, the API isn't set yet. There have been two prototypes so far. The experiences from both have shaped what exists today, but there is still quite a bit left to do. So as I continue down the path of figuring things out, things are almost certainly going to change. **_Expect_** breaking changes in the early releases. I will _strive_ not to publish broken builds and will lean into Semver to identify `alpha`, `beta` and pre-releases.
 
 ## Motivation
 

--- a/cmd/markdown.go
+++ b/cmd/markdown.go
@@ -34,7 +34,6 @@ const (
 )
 
 func createMarkdown(mdParam, planStr string) (*os.File, string, error) {
-	fmt.Printf("mdParam inside createMarkdown is %s\n", mdParam)
 	planMd, err = os.Create(mdParam)
 	if err != nil {
 		log.Errorf("failed to create Markdown: %s\n", err)

--- a/cmd/root.go
+++ b/cmd/root.go
@@ -3,21 +3,17 @@
 package cmd
 
 import (
-	"context"
 	"errors"
 	"fmt"
 	"io"
 	"os"
-	"path/filepath"
 	"runtime"
 	"runtime/debug"
 	"time"
 
-	"github.com/briandowns/spinner"
 	"github.com/charmbracelet/log"
 	"github.com/cli/safeexec"
 	"github.com/fatih/color"
-	"github.com/hashicorp/terraform-exec/tfexec"
 	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"golang.org/x/text/cases"
@@ -98,101 +94,11 @@ var rootCmd = &cobra.Command{
 		files := checkFilesByExtension(workingDir, exts)
 		// we check to see if there are tf or tofu files in the current working directory. If not, we don't call tf.plan
 		if files {
-			if args[0] == "-" {
-				fmt.Println("I'm here!")
-				out = cmd.InOrStdin()
-				content, err := io.ReadAll(out)
+			if len(args) == 0 {
+				planStr, err = createPlan()
 				if err != nil {
-					log.Errorf("unable to read stdIn: %s", err)
+					log.Errorf("Unable to create plan: %s", err)
 				}
-
-				// fmt.Printf("plan output: %s", planStr)
-				mdParam = viper.GetString("mdFile")
-
-				planStr := string(content)
-
-				log.Debug(planStr)
-				fmt.Println("I made it to here.")
-				// Create the plan from Stdin.
-				planMd, mdParam, err = createMarkdown(mdParam, planStr)
-				if err != nil {
-					log.Errorf("Something is not right, %s", err)
-				}
-				// the arg received looks like a file, we try to open it
-			} else if len(args) == 0 {
-				execPath, err := safeexec.LookPath(binary)
-				if err != nil {
-					log.Fatal(
-						"Please ensure either `tofu` or `terraform` are installed and on your $PATH.",
-					)
-					// os.Exit(1)
-				}
-
-				workingDir = filepath.Base(".")
-				// Initialize tf -- NOT terraform init
-				tf, err := tfexec.NewTerraform(workingDir, execPath)
-				if err != nil {
-					log.Fatalf("error calling binary: %s\n", err)
-				}
-
-				// Check for .terraform.lock.hcl -- do not need to do this every time
-				// terraform init | installs providers, etc.
-				// err = tf.Init(context.Background())
-				// if err != nil {
-				//	log.Fatalf("error running Init: %s", err)
-				// }
-
-				// the plan file
-				planPath = viper.GetString("planFile")
-				planOpts := []tfexec.PlanOption{
-					// terraform plan --out planPath (plan.out)
-					tfexec.Out(planPath),
-				}
-
-				// fmt.Printf("plan output: %s", planStr)
-				mdParam = viper.GetString("mdFile")
-
-				log.Debugf("Creating %s plan file %s...", binary, planPath)
-				// terraform plan -out plan.out -no-color
-				spinnerDuration = 100
-				s := spinner.New(spinner.CharSets[14], spinnerDuration*time.Millisecond)
-				s.Suffix = "  Creating the Plan...\n"
-				s.Start()
-
-				_, err = tf.Plan(context.Background(), planOpts...)
-				if err != nil {
-					// binary defined. .tf or .tofu files exist. Still errors. Show me the error
-					log.With("err", err).Errorf("%s returned the follow error", binary)
-					// Edge case exists where we detect .tofu file but terraform was called,
-					// which doesn't support .tofu files. tf.Plan returns error.
-					// There is a condition that exists where .tofu files exist, but terraform
-					// is the binary, this error will occur. But we're not checking _explicitly_
-					// for either .tf or .tofu in files above.
-					// So .tf files _could_ exist, but tf.Plan could fail for
-					// some reason not related to Terraform not finding any .tf files,
-					// making this error inaccurate. Could be nice to identify and
-					// handle this edge case, but Terraform/Tofu do it good enough for now.
-					// if binary == "terraform" {
-					// 	log.Infof("Detected `*.tofu` files, but you've defined %s
-					// as the binary to use in your .tp.toml config file. Terraform does not support `.tofu` files.", binary)
-					// }
-					// We need to exit on this error. tf.Plan actually returns status 1
-					// -- maybe some day we can intercept it or have awareness that it was returned.
-					log.Infof("Check the output of `%s plan` locally. If you believe this is a bug, please report the issue. TPE001.", binary)
-					os.Exit(1)
-				}
-				s.Stop()
-
-				planStr, err = tf.ShowPlanFileRaw(context.Background(), planPath)
-				if err != nil {
-					log.Error(
-						"error internally attempting to create the human-readable Plan: ",
-						err,
-					)
-				}
-
-				log.Debug((planStr))
-
 				// Create the Markdown from the Plan.
 				planMd, mdParam, err = createMarkdown(mdParam, planStr)
 				if err != nil {
@@ -211,6 +117,25 @@ var rootCmd = &cobra.Command{
 					// I'm only human. NFC how you got here. I hope to never have to find out.
 					log.Errorf("If you see this error message, please open a bug. Error Code: TPE002. Error: %s", err)
 				}
+			} else if args[0] == "-" {
+				out = cmd.InOrStdin()
+				content, err := io.ReadAll(out)
+				if err != nil {
+					log.Errorf("unable to read stdIn: %s", err)
+				}
+
+				log.Debugf("plan output: %s", planStr)
+				mdParam = viper.GetString("mdFile")
+
+				planStr := string(content)
+
+				log.Debug(planStr)
+				// Create the plan from Stdin.
+				planMd, mdParam, err = createMarkdown(mdParam, planStr)
+				if err != nil {
+					log.Errorf("Something is not right, %s", err)
+				}
+				// the arg received looks like a file, we try to open it
 			}
 		} else {
 			log.Errorf("No %s files found. Please run this in a directory with %s files present.", cases.Title(language.English).String(binary), cases.Title(language.English).String(binary))

--- a/cmd/tf.go
+++ b/cmd/tf.go
@@ -1,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+
+package cmd
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/charmbracelet/log"
+	"github.com/cli/safeexec"
+	"github.com/hashicorp/terraform-exec/tfexec"
+	"github.com/spf13/viper"
+)
+
+func createPlan() (planStr string, err error) {
+	execPath, err := safeexec.LookPath(binary)
+	if err != nil {
+		log.Fatal(
+			"Please ensure either `tofu` or `terraform` are installed and on your $PATH.",
+		)
+		// os.Exit(1)
+	}
+
+	workingDir = filepath.Base(".")
+	// Initialize tf -- NOT terraform init
+	tf, err := tfexec.NewTerraform(workingDir, execPath)
+	if err != nil {
+		log.Fatalf("error calling binary: %s\n", err)
+	}
+
+	// Check for .terraform.lock.hcl -- do not need to do this every time
+	// terraform init | installs providers, etc.
+	// err = tf.Init(context.Background())
+	// if err != nil {
+	//	log.Fatalf("error running Init: %s", err)
+	// }
+
+	// the plan file
+	planPath = viper.GetString("planFile")
+	planOpts := []tfexec.PlanOption{
+		// terraform plan --out planPath (plan.out)
+		tfexec.Out(planPath),
+	}
+
+	log.Debugf("plan output: %s", planStr)
+	mdParam = viper.GetString("mdFile")
+
+	log.Debugf("Creating %s plan file %s...", binary, planPath)
+	// terraform plan -out plan.out -no-color
+	spinnerDuration = 100
+	s := spinner.New(spinner.CharSets[14], spinnerDuration*time.Millisecond)
+	s.Suffix = "  Creating the Plan...\n"
+	s.Start()
+
+	_, err = tf.Plan(context.Background(), planOpts...)
+	if err != nil {
+		// binary defined. .tf or .tofu files exist. Still errors. Show me the error
+		log.With("err", err).Errorf("%s returned the follow error", binary)
+		// Edge case exists where we detect .tofu file but terraform was called,
+		// which doesn't support .tofu files. tf.Plan returns error.
+		// There is a condition that exists where .tofu files exist, but terraform
+		// is the binary, this error will occur. But we're not checking _explicitly_
+		// for either .tf or .tofu in files above.
+		// So .tf files _could_ exist, but tf.Plan could fail for
+		// some reason not related to Terraform not finding any .tf files,
+		// making this error inaccurate. Could be nice to identify and
+		// handle this edge case, but Terraform/Tofu do it good enough for now.
+		// if binary == "terraform" {
+		// 	log.Infof("Detected `*.tofu` files, but you've defined %s
+		// as the binary to use in your .tp.toml config file. Terraform does not support `.tofu` files.", binary)
+		// }
+		// We need to exit on this error. tf.Plan actually returns status 1
+		// -- maybe some day we can intercept it or have awareness that it was returned.
+		log.Infof("Check the output of `%s plan` locally. If you believe this is a bug, please report the issue. TPE001.", binary)
+		os.Exit(1)
+	}
+	s.Stop()
+
+	planStr, err = tf.ShowPlanFileRaw(context.Background(), planPath)
+	if err != nil {
+		log.Error(
+			"error internally attempting to create the human-readable Plan: ",
+			err,
+		)
+	}
+
+	log.Debug((planStr))
+
+	return planStr, err
+}

--- a/example/extended-workflow-example.md
+++ b/example/extended-workflow-example.md
@@ -11,9 +11,9 @@ Let's grab our example config file and run `gh tp` again:
 
 ```bash
 cp ../gh-tp/.tp.toml .
-gh tp
-2025/03/01 12:36:26 Plan file plan.out was created.
-2025/03/01 12:36:26 Markdown file plan.md was created.
+ gh tp
+✔  Markdown Created...
+✔  Plan Created...
 ```
 
 > [!NOTE]
@@ -97,7 +97,7 @@ Plan: 5 to add, 0 to change, 0 to destroy.
 
 </details>
 
-And we can verify the `plan.md` matches our output above :
+And we can verify the `plan.md` matches our output above:
 
 ```bash
 cat plan.md
@@ -106,77 +106,82 @@ cat plan.md
 ````md
 <details><summary>Terraform Plan</summary>
 
-  ```terraform
+\```terraform
 
-  Terraform used the selected providers to generate the following execution
-  plan. Resource actions are indicated with the following symbols:
-    + create
+Terraform used the selected providers to generate the following execution
+plan. Resource actions are indicated with the following symbols:
 
-  Terraform will perform the following actions:
+- create
 
-    # archive_file.tf_pr will be created
-    + resource "archive_file" "tf_pr" {
-        + id                  = (known after apply)
-        + output_base64sha256 = (known after apply)
-        + output_base64sha512 = (known after apply)
-        + output_md5          = (known after apply)
-        + output_path         = "./tf-pr.tar.gz"
-        + output_sha          = (known after apply)
-        + output_sha256       = (known after apply)
-        + output_sha512       = (known after apply)
-        + output_size         = (known after apply)
-        + source_file         = "./.tf-pr"
-        + type                = "tar.gz"
-      }
+Terraform will perform the following actions:
 
-    # local_file.pet will be created
-    + resource "local_file" "pet" {
-        + content              = (known after apply)
-        + content_base64sha256 = (known after apply)
-        + content_base64sha512 = (known after apply)
-        + content_md5          = (known after apply)
-        + content_sha1         = (known after apply)
-        + content_sha256       = (known after apply)
-        + content_sha512       = (known after apply)
-        + directory_permission = "0777"
-        + file_permission      = "0777"
-        + filename             = "./pet.out"
-        + id                   = (known after apply)
-      }
+# archive_file.tf_pr will be created
 
-    # local_file.uuid will be created
-    + resource "local_file" "uuid" {
-        + content              = (known after apply)
-        + content_base64sha256 = (known after apply)
-        + content_base64sha512 = (known after apply)
-        + content_md5          = (known after apply)
-        + content_sha1         = (known after apply)
-        + content_sha256       = (known after apply)
-        + content_sha512       = (known after apply)
-        + directory_permission = "0777"
-        + file_permission      = "0777"
-        + filename             = "./uuid.out"
-        + id                   = (known after apply)
-      }
+- resource "archive_file" "tf_pr" {
+  - id = (known after apply)
+  - output_base64sha256 = (known after apply)
+  - output_base64sha512 = (known after apply)
+  - output_md5 = (known after apply)
+  - output_path = "./tf-pr.tar.gz"
+  - output_sha = (known after apply)
+  - output_sha256 = (known after apply)
+  - output_sha512 = (known after apply)
+  - output_size = (known after apply)
+  - source_file = "./.tf-pr"
+  - type = "tar.gz"
+    }
 
-    # random_pet.pet will be created
-    + resource "random_pet" "pet" {
-        + id        = (known after apply)
-        + length    = 2
-        + separator = "-"
-      }
+# local_file.pet will be created
 
-    # random_uuid.uuid will be created
-    + resource "random_uuid" "uuid" {
-        + id     = (known after apply)
-        + result = (known after apply)
-      }
+- resource "local_file" "pet" {
+  - content = (known after apply)
+  - content_base64sha256 = (known after apply)
+  - content_base64sha512 = (known after apply)
+  - content_md5 = (known after apply)
+  - content_sha1 = (known after apply)
+  - content_sha256 = (known after apply)
+  - content_sha512 = (known after apply)
+  - directory_permission = "0777"
+  - file_permission = "0777"
+  - filename = "./pet.out"
+  - id = (known after apply)
+    }
 
-  Plan: 5 to add, 0 to change, 0 to destroy.
+# local_file.uuid will be created
 
-  \```
+- resource "local_file" "uuid" {
+  - content = (known after apply)
+  - content_base64sha256 = (known after apply)
+  - content_base64sha512 = (known after apply)
+  - content_md5 = (known after apply)
+  - content_sha1 = (known after apply)
+  - content_sha256 = (known after apply)
+  - content_sha512 = (known after apply)
+  - directory_permission = "0777"
+  - file_permission = "0777"
+  - filename = "./uuid.out"
+  - id = (known after apply)
+    }
 
-  </details>
+# random_pet.pet will be created
+
+- resource "random_pet" "pet" {
+  - id = (known after apply)
+  - length = 2
+  - separator = "-"
+    }
+
+# random_uuid.uuid will be created
+
+- resource "random_uuid" "uuid" {
+  - id = (known after apply)
+  - result = (known after apply)
+    }
+
+Plan: 5 to add, 0 to change, 0 to destroy.
+\```
+
+</details>
 ````
 
 ### We can then apply our Terraform
@@ -201,8 +206,8 @@ Run `gh tp` again, and there are no changes as expected.
 
 ```bash
 gh tp
-2025/02/28 22:58:27 No changes.Your infrastructure matches the configuration.
-2025/02/28 22:58:27 plan.out was created
+✔  Markdown Created...
+✔  Plan Created...
 ```
 
 We can verify that our Markdown contains as much:
@@ -212,17 +217,18 @@ cat plan.md
 ```
 
 ````md
+<details><summary>Terraform Plan</summary>
 
-   <details><summary>Terraform Plan</summary>
+\```terraform
 
-   ```terraform
+No changes. Your infrastructure matches the configuration.
 
-   No Changes. Your Infrastructure matches the configuration.
+Terraform has compared your real infrastructure against your configuration
+and found no differences, so no changes are needed.
 
-   \```
+\```
 
-   </details>
-
+</details>
 ````
 
 > [!NOTE]

